sasmodels/models/octahedron_truncated.py

```diff
@@ -0,0 +1,232 @@
+A square facet crosses the x, y, z directions at distances equal to  t length_a, t length_b and t length_c.
+
+A regular octahedron corresponds to:
+
+.. math::
+    length_a = length_b = length_c
+    t = 1
+
+A regular cuboctahedron shape with 6 squares and 8 triangles corresponds to:
+
+.. math:: 
+    length_a = length_b = length_c
+    t = 1/2
+
+The model contains 4 parameters: length_a, the two ratios b2a_ratio and c2a_ratio and t:
+
+.. math::
+    b2a_ratio = length_b/length_a,
+    c2a_ratio = length_c/length_a,
+    1/2 < t < 1
+
+For a regular shape:
+
+.. math::
+    b2a_ratio = c2a_ratio = 1
+
+Volume of the general shape including truncation is given by:
+
+.. math::
+    V = \frac{4}{3}\, \text{length\_a}^{3}\, \text{b2a\_ratio}\, \text{c2a\_ratio}\,\bigl(1 - 3(1 - t)^{3}\bigr)
+
+The general octahedron is made of eight triangular faces. The three edge lengths
+are:
+
+.. math::
+    A_{\text{edge}}^{2} = \text{length\_a}^{2} + \text{length\_b}^{2},\qquad
+    B_{\text{edge}}^{2} = \text{length\_a}^{2} + \text{length\_c}^{2},\qquad
+    C_{\text{edge}}^{2} = \text{length\_b}^{2} + \text{length\_c}^{2}
+
+For a regular shape (no elongation):
+
+.. math::
+    	{b2a\_ratio} = \text{c2a\_ratio} = 1,\qquad
+    A_{\text{edge}} = B_{\text{edge}} = C_{\text{edge}} = \text{length\_a}\sqrt{2},\qquad
+    	{length\_a} = \text{length\_b} = \text{length\_c} = A_{\text{edge}}/\sqrt{2},
+
+    V = \frac{4}{3}\,\text{length\_a}^{3}\,\bigl(1 - 3(1 - t)^{3}\bigr)
+
+Amplitude of the form factor AP for the fixed orientation of the shape reads
+
+.. math::
+    AP(q,\theta,\phi) = \frac{6}{1 - 3(1 - t)^3}\,(AA + AB + AC)
+
+.. math::
+    AA = \frac{1}{2\,(q_y^2 - q_z^2)\,(q_y^2 - q_x^2)}\Big[(q_y - q_x)\sin\big(q_y(1 - t) - q_x t\big)
+    + (q_y + q_x)\sin\big(q_y(1 - t) + q_x t\big)\Big]
+    + \frac{1}{2\,(q_z^2 - q_x^2)\,(q_z^2 - q_y^2)}\Big[(q_z - q_x)\sin\big(q_z(1 - t) - q_x t\big)
+    + (q_z + q_x)\sin\big(q_z(1 - t) + q_x t\big)\Big]
+
+.. math::
+    AB = \frac{1}{2\,(q_z^2 - q_x^2)\,(q_z^2 - q_y^2)}\Big[(q_z - q_y)\sin\big(q_z(1 - t) - q_y t\big)
+    + (q_z + q_y)\sin\big(q_z(1 - t) + q_y t\big)\Big]
+    + \frac{1}{2\,(q_x^2 - q_y^2)\,(q_x^2 - q_z^2)}\Big[(q_x - q_y)\sin\big(q_x(1 - t) - q_y t\big)
+    + (q_x + q_y)\sin\big(q_x(1 - t) + q_y t\big)\Big]
+
+.. math::
+    AC = \frac{1}{2\,(q_x^2 - q_y^2)\,(q_x^2 - q_z^2)}\Big[(q_x - q_z)\sin\big(q_x(1 - t) - q_z t\big)
+    + (q_x + q_z)\sin\big(q_x(1 - t) + q_z t\big)\Big]
+    + \frac{1}{2\,(q_y^2 - q_z^2)\,(q_y^2 - q_x^2)}\Big[(q_y - q_z)\sin\big(q_y(1 - t) - q_z t\big)
+    + (q_y + q_z)\sin\big(q_y(1 - t) + q_z t\big)\Big]
+
+.. math::
+    q_x = q\,\sin\theta\,\cos\phi, \qquad
+    q_y = q\,\sin\theta\,\sin\phi, \qquad
+    q_z = q\,\cos\theta
+
+θ is the angle between the scattering vector and the z axis of the octahedron (length_c).
+ϕ is the angle between the scattering vector (lying in the xy plane) and the x axis (length_a).
```

Comment:
Other models are using (a, b, c) as the axes for the shape rather than (x, y, z).